gh-115317: Rewrite changelog filter to use vanilla JavaScript

hugovk opened the pull request:
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNN: Summary of the changes made
```

Where: gh-NNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->

Sphinx removed the jQuery library to improve performance, but the changelog filter was still using it. This PR rewrites it to use basic JavaScript. 


<!-- gh-issue-number: gh-115317 -->
* Issue: gh-115317
<!-- /gh-issue-number -->

# Demo

https://cpython-previews--115324.org.readthedocs.build/en/115324/whatsnew/changelog.html

hugovk committed:
Rewrite using vanilla JavaScript

hugovk committed:
Re-add comments

aisk commented on Doc/tools/static/changelog_search.js:
```diff
@@ -1,53 +1,54 @@
             else
-                h2_parent.hide();
-        });
+                h2.parentNode.style.display = 'none';
```
Although there is not a style guide for JavaScript, I think it's better to add the braces here.

hugovk replied:
Added!

hugovk committed:
Add braces

tomasr8 commented on Doc/tools/static/changelog_search.js:
```diff
@@ -1,53 +1,56 @@
+            if (sections_found > 0) {
+                h2.parentNode.style.display = 'block';
+            }
+            else {
```
I think this is the more common formatting for blocks in js
```suggestion
            } else {
```

hugovk replied:
Yeah, I'm going to just run Prettier on this :)

tomasr8 commented on Doc/tools/static/changelog_search.js:
```diff
@@ -1,53 +1,56 @@
-            var h2_parent = $(h2).parent();
-            var sections_found = 0;
+        const h2s = document.querySelectorAll('#changelog h2');
+        for(let h2 of h2s) {
```
`const` is usually preferred if the variable doesn't change
```suggestion
        for(const h2 of h2s) {
```

tomasr8 reviewed (commented):
Just noticed some minor style things ;)

hugovk committed:
Prefer const

Co-authored-by: Tomas R <tomas.roun8@gmail.com>

hugovk committed:
Format with Prettier

hugovk committed:
Use camelCase for function name

For example:

https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/JavaScript#functions

tomasr8 commented on Doc/tools/static/changelog_search.js:
```diff
@@ -1,53 +1,59 @@
+      let sections_found = 0;
+      // find headers for the sections (Core, Library, etc.)
+      const h3s = h2.parentNode.querySelectorAll("h3");
+      for (let h3 of h3s) {
```
One more ;)
```suggestion
      for (const h3 of h3s) {
```

hugovk committed:
Prefer const

Co-authored-by: Tomas R <tomas.roun8@gmail.com>
